src/common/paddle/index.ts

```diff
@@ -190,3 +195,49 @@ export const getTransactionForProviderId = async ({
     })
     .getOne();
 };
+
+export const extractSubscriptionCycle = (
+  items: PaddleSubscriptionEvent['data']['items'],
+) => {
+  const cycle = items?.[0]?.price?.billingCycle?.interval;
+
+  if (!cycle) {
+    return undefined;
+  }
+
+  return cycle === 'year'
+    ? SubscriptionCycles.Yearly
+    : SubscriptionCycles.Monthly;
+};
+
+export const insertClaimableItem = async (
+  con: DataSource | EntityManager,
+  data: SubscriptionNotification | SubscriptionCreatedNotification,
+) => {
+  const customer = await paddleInstance.customers.get(data.customerId);
+
+  const existingEntries = await con.getRepository(ClaimableItem).find({
+    where: {
+      email: customer.email,
+      claimedAt: IsNull(),
+    },
+  });
+
+  await con.getRepository(ClaimableItem).insert({
+    type: ClaimableItemTypes.Plus,
+    email: customer.email,
+    flags: {
+      cycle: extractSubscriptionCycle(data.items),
+      createdAt: data.startedAt,
+      subscriptionId: data.id,
+      provider: SubscriptionProvider.Paddle,
+      status: UserSubscriptionStatus.Active,
+    },
+  });
+
+  if (existingEntries.length > 0) {
+    throw new Error(
+      `User ${customer.email} already has a claimable subscription`,
+    );
+  }
```

Comment:
Should we move this above the `insert`, because now it will insert another row and then throw error.